great_expectations/datasource/fluent/config.py

```diff
@@ -152,6 +157,27 @@ def update_datasources(self, datasources: Dict[str, Datasource]) -> None:
         datasources_as_dict.update(datasources)
         self.fluent_datasources = list(datasources_as_dict.values())
 
+    def pop(self, datasource_name: str, default: T = _MISSING) -> Datasource | T:  # type: ignore[assignment] # sentinel value is never returned
```

Comment:
Can you explain this sentinel value a bit more?

Reply:
Yes, this technique is to help distinguish whether or not a value was passed to `default`.
`_MISSING = object()` creates an object with a unique reference that no user will ever pass by accident.

Whereas if I had a default of `None` there's no way for me to tell whether or not it's the default `None` or a user-provided `None`.

Another similar technique might be to assign `_MISSING` to something like a UUID or some magic string.

This all assumes I don't want to use `*args`, that's another technique I could use here.